ci/docker.sh

```diff
@@ -37,32 +37,34 @@ if [ -z "${docker_hub_username:-}" ]; then
   fn_die "ERROR: DOCKER_HUB_USERNAME variable is not set. Exiting ..."
 fi
 
-docker_tag=""
+docker_tag_full=""
 if [ "${is_a_release}" = "true" ]; then
-  docker_tag="${github_ref_name}"
+  docker_tag_full="${github_ref_name}"
 fi
 
 # Building and publishing docker image
-if [ -n "${docker_tag:-}" ]; then
-  log_info "=== Building Docker image: ${docker_hub_org}/${docker_image_build_name}:${docker_tag} ==="
-  docker build --build-arg PROFILE=production -f "${docker_file_path}" -t "${docker_hub_org}/${docker_image_build_name}:${docker_tag}" .
+if [ -n "${docker_tag_full:-}" ]; then
+  log_info "=== Building Docker image: ${docker_hub_org}/${docker_image_build_name}:${docker_tag_full} ==="
+  docker build --build-arg PROFILE=production -f "${docker_file_path}" -t "${docker_hub_org}/${docker_image_build_name}:${docker_tag_full}" .
 
   # Publishing to DockerHub
   log_info "=== Publishing Docker image(s) on Docker Hub ==="
   echo "${docker_hub_token}" | docker login -u "${docker_hub_username}" --password-stdin
 
   # Docker image(s) tags for PROD vs DEV release
   if [ "${prod_release}" = "true" ]; then
-    publish_tags=("${docker_tag}" "latest")
+    docker_tag_node=$(cut -d'-' -f1 <<< $docker_tag_full)
+    publish_tags=("${docker_tag_full}" "${docker_tag_node}" "latest")
   elif [ "${dev_release}" = "true" ]; then
-    publish_tags=("${docker_tag}" "dev")
+        docker_tag_node=$(cut -d'-' -f1 <<< $docker_tag_full)-$(cut -d'-' -f3- <<< $docker_tag_full)
```

Comment:
These cut commands will lead to the following value if TAG is 1.1.1-2.2.2 for example that is being introduced push by tags regex

`docker_tag_node=1.1.1-` which I think not what we want?

If tag is 1.1.1 then `docker_tag_node=1.1.1-2.2.2` 
It seems like the same is happening for PROD_RELEASE? Why not to use the same approach then? 

docker_full_tag=github_ref_name (= 1.1.1-2.2.2)

```
docker_tag_node="$(cut -d '-' -f1 <<< "${docker_tag_full}")"   (= 1.1.1)
```

Reply:
Modified as per your suggestion.